### PR TITLE
Fix IsPermanent to account for wrapped errors

### PR DESCRIPTION
**Pull request by joe-elliott**
**Description:**
Currently IsPermanent does not correctly test for wrapped errors.  We noticed this when the collector was incorrectly retrying otlp grpc FailedPrecondition status:

```
level=error msg="Exporting failed. No more retries left. Dropping data." component=tempo component_kind=exporter component_type=otlp component_name=otlp error="max elapsed time expired failed to push trace data via OTLP exporter: Permanent error: fatal error sending to server" dropped_items=100
```
This occurs b/c the error is wrapped and the `IsPermanent` function test does not correctly detect wrapped errors.

Error wrapping:

https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/otlpexporter/otlp.go#L79

Go Playground with examples:

https://play.golang.org/p/kVBm3SesDBu

**Testing:**
Added a simple test to detect a wrapped error.

### CODE REVIEW

**Review by mwear — APPROVED** [review]
I ran into this and was preparing a fix as well. The OTLP exporter potentially rewraps permanent errors, which end up getting retried, for example: https://github.com/open-telemetry/opentelemetry-collector/blob/9de6dd77eeaef5fd2557a5080df14e9b083de938/exporter/otlpexporter/otlp.go#L91

Can confirm that this fixes that issue.

File: consumer/consumererror/permanenterror_test.go [hunk 1]
```diff
@@ -27,9 +28,12 @@ func TestPermanent(t *testing.T) {
...
+
+	err = fmt.Errorf("%w", err)
+	require.True(t, IsPermanent(err))
 }
 
 func TestIsPermanent_NilError(t *testing.T) {
 	var err error
 	require.False(t, IsPermanent(err))
-}
+}
```
Comment (mwear) [inline consumer/consumererror/permanenterror_test.go:39]: The lint task is failing due to missing a new line here
```suggestion
}

```

**Review by jpkrohling — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
Thanks for fixing this!

**Review by carlosalberto — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@joe-elliott please fix the tests, I tried to re-run them multiple times and same error. Maybe a rebase would help.

**Comment by joe-elliott** [comment]
Wow, yeah.  I was way further behind than I realized.  Rebased and pushed.

**Pull request merged**